packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1469,16 +1474,9 @@ export class ImportResolver {
         return this._cachedTypeshedThirdPartyPackageRoots!;
     }
 
-    private _getTypeshedSubdirectory(isStdLib: boolean, execEnv: ExecutionEnvironment, importFailureInfo: string[]) {
-        // See if we have it cached.
-        if (isStdLib) {
-            if (this._cachedTypeshedStdLibPath !== undefined) {
-                return this._cachedTypeshedStdLibPath;
-            }
-        } else {
-            if (this._cachedTypeshedThirdPartyPath !== undefined) {
-                return this._cachedTypeshedThirdPartyPath;
-            }
+    private _getTypeshedRoot(execEnv: ExecutionEnvironment, importFailureInfo: string[]) {
+        if (this._cachedTypeshedRoot) {
```

Comment:
You should probably check for "!== undefined" here because an empty string is a valid cached value.

Reply:
Yes, this doesn't mirror the other cache lines; will fix.